cmd/omniwitness/monolith.go

```diff
@@ -54,6 +58,23 @@ func main() {
 	flag.Parse()
 	ctx := context.Background()
 
+	if *metricsAddr == "" {
+		mf := monitoring.InertMetricFactory{}
+		omniwitness.InitMetrics(mf)
```

Comment:
I can see there could be a growing tree of recursive `InitMetrics` calls, I wonder if it might be cleaner to set the runtime-wide implementation via a func in `monitoring` - e.g. `monitoring.SetMetricFactory(mf)`, and then have packages which want to provide metrics create them with their own `Once` func at an appropriate time (e.g. in a `New...()` or a package func, etc.)?

wdyt?

Reply:
Yeah I really didn't like this either. I was starting to look into whether `google/wire` solves this but I'm not convinced it does, and that project hasn't been updated in a year which is a bit suspect too.

I've gone with the static singleton approach. I don't love this, but it's evidently superior to the other options.